scripts/build/arch/powerpc.sh

```diff
@@ -26,3 +26,44 @@ CT_DoArchTupleValues () {
         CT_ARCH_CC_EXTRA_CONFIG="--enable-e500_double"
     fi
 }
+#------------------------------------------------------------------------------
+# Get multilib architecture-specific target
+# Usage: CT_DoArchMultilibTarget "multilib flags" "target tuple"
+CT_DoArchMultilibTarget ()
+{
+    local multi_flags="${1}"
+    local target="${2}"
+
+    local m32=false
+    local m64=false
+    local mlittle=false
+    local mbig=false
+
+    for m in $multi_flags; do
+        case "$m" in
+            -m32)     m32=true ;;
+            -m64)     m64=true ;;
+            -mbig)    mbig=true ;;
+            -mlittle) mlittle=true ;;
+        esac
+    done
+
+    # Fix up bitness
+    case "${target}" in
+        powerpc-*)      $m64 && target=${target/#powerpc-/powerpc64-} ;;
+        powerpcle-*)    $m64 && target=${target/#powerpcle-/powerpc64le-} ;;
+        powerpc64-*)    $m32 && target=${target/#powerpc64-/powerpc-} ;;
+        powerpc64le-*)  $m32 && target=${target/#powerpc64le-/powerpcle-} ;;
+    esac
+
+    # Fix up endianness
+    case "${target}" in
+        powerpc-*)      $mlittle && target=${target/#powerpc-/powerpcle-} ;;
+        powerpcle-*)    $mbig && target=${target/#powerpcle-/powerpc-} ;;
+        powerpc64-*)    $mlittle && target=${target/#powerpc64-/powerpc64le-} ;;
+        powerpc64le-*)  $mbig && target=${target/#powerpc64le-/powerpc64-} ;;
+    esac
+
+    # return the target
+    echo "${target}"
```

Comment:
It took me a while to understand that `CT_DoArchMultilibTarget` can be
called multiple times, with different `multi_flags` each time, and that
we would only get one of each kind with each call, i.e. that we can't
have both `-m32` and `-m64` in the same call. Right?

Please make that explicit in the commit log. Otherwise, one has to look
at the following commits to understand how/when it is called.